src/opensearch/client.py

```
@@ -120,29 +128,55 @@ def _initialize_client_single_mode() -> OpenSearch:
     try:
         # Get connection parameters from environment variables
         opensearch_url = os.getenv('OPENSEARCH_URL', '').strip()
-        if not opensearch_url:
-            raise ConfigurationError(
-                'OPENSEARCH_URL environment variable must be set for single mode'
-            )
-
         opensearch_username = os.getenv('OPENSEARCH_USERNAME', '').strip()
         opensearch_password = os.getenv('OPENSEARCH_PASSWORD', '').strip()
         opensearch_no_auth = os.getenv('OPENSEARCH_NO_AUTH', '').lower() == 'true'
         iam_arn = os.getenv('AWS_IAM_ARN', '').strip()
         profile = os.getenv('AWS_PROFILE', '').strip()
-
-        # Check if serverless
         is_serverless_mode = os.getenv('AWS_OPENSEARCH_SERVERLESS', '').lower() == 'true'
         opensearch_timeout_str = os.getenv('OPENSEARCH_TIMEOUT', '').strip()
         opensearch_timeout = int(opensearch_timeout_str) if opensearch_timeout_str else None
-
-        # Get SSL verification setting (single mode uses environment variable)
         ssl_verify = os.getenv('OPENSEARCH_SSL_VERIFY', 'true').lower() != 'false'
+        aws_access_key_id = None
+        aws_secret_access_key = None
+        aws_session_token = None
+
+        # Check if header auth is enabled and update variables accordingly
+        use_header_auth = os.getenv('OPENSEARCH_HEADER_AUTH', '').lower() == 'true'
+        if use_header_auth:
+            header_auth = _get_auth_from_headers()
+            header_url = header_auth.get('opensearch_url')
+            if header_url:
+                opensearch_url = header_url
+            header_service = header_auth.get('aws_service_name')
+            if header_service:
```

should we throw warning/error for invalid service?

---

I think it is okay, we will default to `es` if an invalid request is sent